py/dml/codegen.py

```diff
@@ -1158,18 +1158,20 @@ def check_designated_initializers(site, etype, init_asts):
             duplicates.add(field)
         else:
             remaining.remove(field)
-    if (duplicates or bad_fields or remaining):
+    if (duplicates or bad_fields or (not allow_partial and remaining)):
         raise EDATAINIT(site,
                         ''.join("\nduplicate initializer for "
                                 + f"member '{field}'"
                                 for field in duplicates)
                         + ''.join(f"\n'{field}' is not a member of "
                                   + f"{etype}"
                                   for field in bad_fields)
-                        + ("\nmissing initializer(s) for field(s) "
+                        + ("\nmissing initializer(s) for member(s) "
                            + ", ".join(f"'{field}'"
                                        for field in remaining)
-                           if remaining else '')
+                           + "\n(partial initialization must be explicit "
+                           + "through trailing '...', e.g. '{.a = a, ...}')"
+                           if not allow_partial and remaining else '')
```

Comment:
Consider `Blah blah.` instead of `(blah blah)`

Reply:
I did! I picked `(blah blah)` so the user could not confuse this string with a sub-error, as missing members, duplicate members, and bad members sub-errors are presented through each category being on their own line.